db/compaction_picker_universal.cc

```diff
@@ -342,8 +342,6 @@ Compaction* UniversalCompactionPicker::PickCompaction(
       assert(f->smallest_seqno <= f->largest_seqno);
       if (is_first) {
         is_first = false;
-      } else {
-        assert(prev_smallest_seqno > f->largest_seqno);
```

Comment:
Yes. It is only used for Levels other that L0, but if there is 1 (or more than 1) file on L0, the initial value of prev_smallest_seqno of the next for loop should be the smallest_seqno of the (last) L0 file and is_first should be false.